modtree/modtree
# feat: implement module node states

### Summary of changes

Add node styling for our 5 module states, and make the graph update styling upon detecting changes. The following actions may trigger a style change:

- Update modules done
- Update modules doing
- Remove node from graph
- Add node to graph

### Testing

For add/remove:

1. Remove all nodes from a graph.
2. Add CS2030. It should be red.
3. Add CS1010. It should be gray, and CS2030 should now be gray as well. 
4. Remove CS1010. CS2030 should now be red again.

For modules done/doing:

1. Remove all nodes from a graph.
2. Add CS2030. It should be red.
3. Mark CS2030 as done. It should be green, with slight fade (opacity 50%)
4. Add CS1010. It should be black.
5. Mark CS1010 as done. It should be green, with slight fade (opacity 50%)
6. Repeat this process, but setting CS1010 and CS2030 to doing instead.
